### PR TITLE
Instantiate credentials using `ibm-credentials.env` 

### PR DESCRIPTION
**Pull request by ehdsouza**
The SDK should search for the credentials file in 3 locations (in order):

The file path specified by an environment variable called IBM_CREDENTIALS_FILE
The user's home directory
Mac/Linux: $HOME
Windows: %HOMEDRIVE%%HOMEPATH% or %USERPROFILE%
The top-level of the project directory

### CODE REVIEW

File: watson_developer_cloud/watson_service.py [hunk 1]
```diff
@@ -259,6 +265,8 @@ def __init__(self, vcap_services_name, url, username=None, password=None,
         elif iam_access_token is not None or iam_apikey is not None:
             self.set_token_manager(iam_apikey, iam_access_token, iam_url)
 
+        self.load_from_credential_file(vcap_services_name)
```
Comment (lpatino10) [inline watson_developer_cloud/watson_service.py]: Based on the team discussion yesterday morning, we decided to go with what's essentially the display name, e.g. Visual Recognition. You'll need to get that value and search for the snake-cased form (`visual_recognition`) in your `_set_credential_based_on_type` function.

File: watson_developer_cloud/watson_service.py [hunk 2]
```diff
@@ -268,7 +268,7 @@ def __init__(self, vcap_services_name, url, username=None, password=None,
             self.set_token_manager(iam_apikey, iam_access_token, iam_url)
 
         # 2. Credentials from credential file
-        if not display_name and not self.username and not self.token_manager:
```
Comment (lpatino10) [inline watson_developer_cloud/watson_service.py]: Definitely a good catch here

**Comment by lpatino10** [comment]
All still looks good with the latest commit!

File: watson_developer_cloud/watson_service.py (hunk 2 above)
Comment (ehdsouza) [reply watson_developer_cloud/watson_service.py]: He he

**Pull request merged**